Source/DataSources/KmlDataSource.js

```diff
@@ -1846,6 +2342,221 @@ define([
+                    networkLink.updating = false;
+                    networkLinks.remove(networkLink.id);
+                    return;
+                }
+                networkLink.cookie = defaultValue(queryStringValue(networkLinkControl, 'cookie', namespaces.kml), '');
+                minRefreshPeriod = defaultValue(queryNumericValue(networkLinkControl, 'minRefreshPeriod', namespaces.kml), 0);
+            }
+
+            var now = JulianDate.now();
+            var refreshMode = networkLink.refreshMode;
+            if (refreshMode === RefreshMode.INTERVAL) {
+                if (defined(networkLinkControl)) {
+                    networkLink.time = Math.max(minRefreshPeriod, networkLink.time);
+                }
+            } else if (refreshMode === RefreshMode.EXPIRE) {
+                var expires;
+                if (defined(networkLinkControl)) {
+                    expires = queryStringValue(networkLinkControl, 'expires', namespaces.kml);
+                }
+                if (defined(expires)) {
+                    try {
+                        var date = JulianDate.fromIso8601(expires);
+                        var diff = JulianDate.secondsDifference(date, now);
+                        if (diff > 0 && diff < minRefreshPeriod) {
+                            JulianDate.addSeconds(now, minRefreshPeriod, date);
+                        }
+                        networkLink.time = date;
+                    } catch (e) {
+                        console.log('KML - NetworkLinkControl expires is not a valid date');
+                        remove = true;
+                    }
+                } else {
+                    console.log('KML - refreshMode of onExpire requires the NetworkLinkControl to have an expires element');
+                    remove = true;
+                }
+            }
+
+            var networkLinkEntity = networkLink.entity;
+            var entityCollection = dataSource._entityCollection;
+            var newEntities = newEntityCollection.values;
+
+            function removeChildren(entity) {
+                entityCollection.remove(entity);
+                var children = entity._children;
+                var count = children.length;
+                for(var i=0;i<count;++i) {
+                    removeChildren(children[i]);
+                }
+            }
+
+            // Remove old entities
+            entityCollection.suspendEvents();
+            var entitiesCopy = entityCollection.values.slice();
+            for (var i=0;i<entitiesCopy.length;++i) {
+                var entityToRemove = entitiesCopy[i];
+                if (entityToRemove.parent === networkLinkEntity) {
+                    entityToRemove.parent = undefined;
+                    removeChildren(entityToRemove);
+                }
+            }
+            entityCollection.resumeEvents();
+
+            // Add new entities
+            entityCollection.suspendEvents();
```

Comment:
No that was in another spot on the initial loading so I could compute availability. That is no longer needed. The entities don't update if I don't have these calls here. It may be a bug somewhere in the Entity API.